WarshipGame/Assets/Scripts/Gameplay/Camera/CameraController.cs

```diff
@@ -0,0 +1,175 @@
+
+public class CameraController : MonoBehaviour
+{
+    [SerializeField] private CinemachineVirtualCamera CMVirtualCamera;
+    [SerializeField] private Camera Cam;
+
+    [Header("Panning")]
+    
+    [Tooltip("The speed in which the player can drag the camera across the game scene.")]
+    [SerializeField] private float PanningSpeed = 10f;
+    [Tooltip("The amount of Hexes in the X-axis on the map.")]
+    [SerializeField] private float MapSizeX = 8f;
+    [Tooltip("The amount of Hexes in the Z-axis on the map.")]
+    [SerializeField] private float MapSizeZ = 8f;
+    private Vector3 _cameraOrigin;
+    private Vector3 _dragOrigin;
+    private Vector3 _dragDirection;
+    
+    [Header("Zooming")]
+    
+    [Tooltip("The FOV of the camera will change to this value once a focus point is selected.")]
+    [SerializeField] private float FocusValue = 30f;
+    private GameObject _focusGameObject;
+    private float _focusBoundaries;
+    private bool _isFocussing;
+    
+    [Space]
+    [Tooltip("The Cameras FOV will always be in between these 2 values.")]
+    [SerializeField] private Vector2 MaxZoom = new(35f,60f);
+    [Tooltip("The Speed in which the player can zoom the camera FOV.")]
+    [SerializeField] private float ZoomSpeed = 9;
+    [Tooltip("The standard zoomFOV for this camera controller.")] 
+    [SerializeField] private float DefaultFOVValue = 45;
+    private float _currentFOVValue;
+    
+    private void Awake()
+    {
+        MapSizeX *= 0.866f;
+        MapSizeZ = MapSizeZ * 0.866f - (0.866f / 2);
+        
+        _currentFOVValue = DefaultFOVValue;
+        CMVirtualCamera.m_Lens.FieldOfView = _currentFOVValue;
+    }
+
+    private void FixedUpdate()
+    {
+        PanCamera();
+
+        if (Input.GetAxis("Mouse ScrollWheel") != 0 && !_isFocussing)
+        {
+            ZoomCamera();
+        }
+    }
+
+    /// <summary>
+    /// This function makes it possible to drag the camera across the
+    /// play field between a certain boundary
+    /// </summary>
+    private void PanCamera()
+    {
+        if (Input.GetMouseButtonDown(0))
+        {
+            _dragOrigin = Input.mousePosition;
+            _cameraOrigin = CMVirtualCamera.transform.position;
+        }
+
+        if (!Input.GetMouseButton(0)) return;
+        if (Input.GetAxis("Mouse X") == 0 && Input.GetAxis("Mouse Y") == 0) return;
+        
+        _dragDirection = Input.mousePosition;
+        Vector3 screenDelta = _dragOrigin - _dragDirection;
+
+        Vector2 screenSize = ScaleScreenToWorldSize(Cam.aspect, Cam.orthographicSize, Cam.scaledPixelWidth, Cam.scaledPixelHeight, screenDelta.x, screenDelta.y);
+
+        Vector3 move = new Vector3(screenSize.x + screenSize.y, 0f, screenSize.y - screenSize.x);
+        move = Quaternion.Euler(0,-90,0) * move;
+        Vector3 camPosMove = new Vector3(_cameraOrigin.x + move.x * Time.deltaTime * PanningSpeed, _cameraOrigin.y, _cameraOrigin.z + move.z * Time.deltaTime * PanningSpeed);
```

Comment:
Because it is already defined as a Vector3 we don't need to say we are making a new Vector3 just new